cache.go

```diff
@@ -224,15 +256,18 @@ func (c *Cache) promote(item *Item) {
 func (c *Cache) worker() {
 	defer close(c.control)
 	dropped := 0
+	promoteItem := func(item *Item) {
+		if c.doPromote(item) && c.size > c.maxSize {
+			dropped += c.gc()
+		}
+	}
```

Comment:
I added this just to avoid repetition below, but if you'd rather avoid the overhead of a function call on line 270, it could be repeated inline on line 291 instead.